api/contrib/envoy/extensions/matching/input_matchers/hyperscan/v3alpha/hyperscan.proto

```diff
@@ -0,0 +1,83 @@
+syntax = "proto3";
+
+package envoy.extensions.matching.input_matchers.hyperscan.v3alpha;
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.matching.input_matchers.hyperscan.v3alpha";
+option java_outer_classname = "HyperscanProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/matching/input_matchers/hyperscan/v3alpha";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Hyperscan matcher]
+// [#extension: envoy.matching.input_matchers.hyperscan]
+
+// `Hyperscan <https://github.com/intel/hyperscan>`_ regex matcher. The engine of the matcher uses
+// hybrid automata techniques to allow simultaneous matching of large numbers of regular
+// expressions across streams of data.
+//
+// The matcher follows PCRE pattern syntax, and the regex string must adhere to the documented
+// `pattern support <https://intel.github.io/hyperscan/dev-reference/compilation.html#pattern-support>`_.
+message Hyperscan {
```

Comment:
I would add a note here, saying that PCRE syntax is not compatible with RE2 matcher, and depending on configured match patterns, swapping regex engines may cause match rules to no longer be valid.

Reply:
Ok, I will add a comment.

By the way, we are introducing Hyperscan into Envoy as an input matcher instead of a regex engine. Users are not expected to switch the regex engine since there in fact has no regex engine choice in Envoy excepting RE2. You may see from the [example usage](https://github.com/envoyproxy/envoy/pull/19473#issuecomment-1054281852) is different from the one using RE2.

Reply:
Oh yes, sorry, what you are saying makes sense. Sorry about this again, I think my suggestion does not make sense. I think we should remove it.